token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -800,13 +806,269 @@ fn process_allow_balance_credits(
         account_info_iter.as_slice(),
     )?;
 
-    let mut ct_token_account = token_account.get_extension_mut::<ConfidentialTransferAccount>()?;
-    ct_token_account.approved()?;
-    ct_token_account.allow_balance_credits = allow_balance_credits.into();
+    let mut confidential_transfer_account =
+        token_account.get_extension_mut::<ConfidentialTransferAccount>()?;
+    confidential_transfer_account.approved()?;
+    confidential_transfer_account.allow_balance_credits = allow_balance_credits.into();
 
     Ok(())
 }
 
+/// Processes an [WithdrawWithheldTokensFromMint] instruction.
+fn process_withdraw_withheld_tokens_from_mint(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    proof_instruction_offset: i64,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let mint_account_info = next_account_info(account_info_iter)?;
+    let dest_account_info = next_account_info(account_info_iter)?;
+    let instructions_sysvar_info = next_account_info(account_info_iter)?;
+    let authority_info = next_account_info(account_info_iter)?;
+    let authority_info_data_len = authority_info.data_len();
+
+    let mut mint_data = mint_account_info.data.borrow_mut();
+    let mut mint = StateWithExtensionsMut::<Mint>::unpack(&mut mint_data)?;
+
+    // mint must be extended for fees
+    {
+        let fee_mint = mint.get_extension::<TransferFeeConfig>()?;
+        let withdraw_withheld_authority =
+            Option::<Pubkey>::from(fee_mint.withdraw_withheld_authority)
+                .ok_or(TokenError::NoAuthorityExists)?;
+        Processor::validate_owner(
+            program_id,
+            &withdraw_withheld_authority,
+            authority_info,
+            authority_info_data_len,
+            account_info_iter.as_slice(),
+        )?;
+    } // free `fee_mint` to borrow `confidential_transfer_mint` as mutable
+
+    let confidential_transfer_mint = mint.get_extension_mut::<ConfidentialTransferMint>()?;
+
+    // basic checks for the destination account - must be extended for confidential transfers
+    let mut dest_account_data = dest_account_info.data.borrow_mut();
+    let mut dest_account = StateWithExtensionsMut::<Account>::unpack(&mut dest_account_data)?;
+    if dest_account.base.mint != *mint_account_info.key {
+        return Err(TokenError::MintMismatch.into());
+    }
+    if dest_account.base.is_frozen() {
+        return Err(TokenError::AccountFrozen.into());
+    }
+    let mut dest_confidential_transfer_account =
+        dest_account.get_extension_mut::<ConfidentialTransferAccount>()?;
+
```

Comment:
let's add an approved check here as well, as an non-approved account isn't allowed to hold any confidential tokens
```suggestion
    dest_confidential_transfer_account.approved()?;
```